pnpm/src/main.ts

```diff
@@ -200,7 +200,7 @@ export async function main (inputArgv: string[]): Promise<void> {
     const filterResults = await filterPackagesFromDir(wsDir, filters, {
       engineStrict: config.engineStrict,
       nodeVersion: config.nodeVersion ?? config.useNodeVersion,
-      patterns: cliOptions['workspace-packages'] ?? config.workspacePackagePatterns,
+      patterns: config.workspacePackagePatterns ?? ['.', '**'],
```

Comment:
This is something I noticed in one of the other PRs too. The `patterns` field here could accidentally be `undefined`, but TypeScript never caught this because `cliOptions` is `Record<string, any>`.

I think this only comes up in tests where `workspaceDir` is directly passed rather than through the `findWorkspaceDir` function. In production, the `findWorkspaceDir` verifies `pnpm-workspace.yaml` exists, so `config.workspacePackagePatterns` is defined.

I added `?? ['.', '**']` to match the existing behavior, but I think we should default to `?? []` in this case so broken tests fail.